src/providers/analytics/provider.ts

```diff
@@ -34,49 +44,64 @@ export class IndexAnalyticsProvider implements AnalyticsProvider {
     private readonly coingeckoService: CoinGeckoService,
   ) {}
 
-  async getAnalytics(address: string): Promise<IndexAnalytics> {
-    const { baseCurrency, coingeckoService, provider } = this
+  async getAnalytics(
+    address: string,
+    options: IndexAnalyticsOptions = {
+      includeTotalSupply: true,
+      include24hrVolume: true,
+    },
+  ): Promise<IndexAnalytics> {
+    const { baseCurrency, chainId, coingeckoService, provider } = this
     const marketCapProvider = new IndexMarketCapProvider(
       provider,
       coingeckoService,
     )
     const navProvider = new IndexNavProvider(provider, coingeckoService)
-    const supplyProvider = new IndexSupplyProvider(provider)
-    // For now we can save time by always assuming Ethereum
-    const chainId = 1
-    // Token data fetched statically to avoid lengthy contract interactions
-    const token = TokenData[address.toLowerCase()]
-    const totalSupply = await supplyProvider.getSupply(address)
-    const coingeckoRes = await coingeckoService.getTokenPrice({
+
+    const supplyPromise = options.includeTotalSupply
+      ? new IndexSupplyProvider(provider).getSupply(address)
+      : Promise.resolve(null)
+    const marketCapPromise = marketCapProvider.getMarketCap(address)
+    const navPricePromise = navProvider.getNav(address)
+    const coingeckoPromise = coingeckoService.getTokenPrice({
       address,
       chainId,
       baseCurrency,
       include24hrChange: true,
-      include24hrVol: true,
+      include24hrVol: options.include24hrVolume ?? false,
```

Comment:
default should be `true`, I'd say

Reply:
I was thinking of it as: "if you're choosing to provide specific options (via the extra parameter), that we'd by default pull the bare minimum amount of data + any items that are specified." e.g. a user that specifies `{ includeTotalSupply: true }` will now also get this data if we change this to `?? true`.

Think we should explicitly opt out of this getting pulled? (A user would need to specify `{ include24hrVolume: false }` to suppress it in the request to CG) Should that behavior apply to the other options in this object?

Regardless of approach, it should be consistent IMO (default enabled or default disabled)

Reply:
🤝